buildkite/resource_cluster_test.go

```diff
@@ -9,105 +9,130 @@ import (
+				CheckDestroy:             testAccCheckClusterDestroy,
+				Steps: []resource.TestStep{
+					{
+						Config: originalConfig,
+						Check: resource.ComposeAggregateTestCheckFunc(
+							testAccCheckClusterExists("buildkite_cluster.foo", &c),
+							testAccCheckClusterRemoteValues(&c, fmt.Sprintf("%s_test_cluster", randName)),
+							resource.TestCheckResourceAttr("buildkite_cluster.foo", "name", fmt.Sprintf("%s_test_cluster", randName)),
+							resource.TestCheckResourceAttrSet("buildkite_cluster.foo", "id"),
+							resource.TestCheckResourceAttrSet("buildkite_cluster.foo", "uuid"),
+						),
+					},
+					{
+						Config: newConfig,
+						Check: resource.ComposeAggregateTestCheckFunc(
+							testAccCheckClusterExists("buildkite_cluster.foo", &c),
+							testAccCheckClusterRemoteValues(&c, fmt.Sprintf("%s_test_cluster", randNameUpdated)),
+							resource.TestCheckResourceAttr("buildkite_cluster.foo", "name", fmt.Sprintf("%s_test_cluster", randNameUpdated)),
+							resource.TestCheckResourceAttrSet("buildkite_cluster.foo", "id"),
+							resource.TestCheckResourceAttrSet("buildkite_cluster.foo", "uuid"),
+						),
+					},
+				},
+			})
+		}
+		t.Run("with basic settings", func(t *testing.T) {
+			testCase(t, basic(randName), basic(randNameUpdated))
+		})
 	})
-}
 
-func TestAccCluster_Import(t *testing.T) {
-	resName := acctest.RandString(13)
-	t.Parallel()
-	var c = new(clusterResourceModel)
-
-	resource.Test(t, resource.TestCase{
-		PreCheck:                 func() { testAccPreCheck(t) },
-		ProtoV6ProviderFactories: protoV6ProviderFactories(),
-		CheckDestroy:             testAccCheckClusterDestroy,
-		Steps: []resource.TestStep{
-			{
-				Config: testAccClusterBasic(resName),
-				Check: resource.ComposeAggregateTestCheckFunc(
-					testAccCheckClusterExists("buildkite_cluster.foo", c),
-					resource.TestCheckResourceAttr("buildkite_cluster.foo", "name", fmt.Sprintf("%s_test_cluster", resName)),
-				),
-			},
-			{
-				ResourceName: "buildkite_cluster.foo",
-				ImportStateIdFunc: func(s *terraform.State) (string, error) {
-					return c.UUID.ValueString(), nil
+	t.Run("Imports a Cluster", func(t *testing.T) {
+		randName := acctest.RandString(5)
+		check := resource.ComposeAggregateTestCheckFunc(
+			testAccCheckClusterExists("buildkite_cluster.foo", &c),
+			resource.TestCheckResourceAttr("buildkite_cluster.foo", "name", fmt.Sprintf("%s_test_cluster", randName)),
+		)
+
+		testCase := func(t *testing.T, config string) {
+			resource.Test(t, resource.TestCase{
+				PreCheck:                 func() { testAccPreCheck(t) },
+				ProtoV6ProviderFactories: protoV6ProviderFactories(),
+				CheckDestroy:             testAccCheckClusterDestroy,
+				Steps: []resource.TestStep{
+					{
+						Config: config,
+						Check:  check,
+					},
+					{
+						ResourceName: "buildkite_cluster.foo",
+						ImportStateIdFunc: func(s *terraform.State) (string, error) {
+							return c.UUID.ValueString(), nil
```

Comment:
Oh I didn't realise cluster is using UUID for importing. We'll need to change this for v1 to use ID like everything else

Reply:
Agreed, but that shouldn't be done in this PR.

Reply:
No thats right. I added SUP-1380